hxrsnd/sndsystem.py

```diff
@@ -385,10 +385,19 @@ def energy(self, E):
                          tth=self.tth.name, th1=self.th1.name,
                          th2=self.th2.name, theta=theta, half_theta=theta/2))
 
-        # Set the position of the motors
-        status_tth = self.tth.move(theta, wait=False)
-        status_th1 = self.th1.move(theta/2, wait=False)
-        status_th2 = self.th2.move(theta/2, wait=False)
+        motors = [self.tth, self.th1, self.th2]
+        move_pos = [theta, theta/2, theta/2]
+
+        # Check that we can move all the motors
+        for motor in motors:
+            try:
+                motor.check_status()
+            except Exception as e:
+                err = "Motor {0} got an exception: {1}".format(motor.name, e)
+                logger.error(err)
+                raise e
+
+        status = [motor.move(pos) for move, pos in zip(motors, move_pos)]
```

Comment:
Maybe move this into a `set_energy` function. That way you can return the statuses and you can wait on them. Otherwise, I think we are screwed if it is all in the `energy.setter` because you can't `status = snd.energy = 4500` and `status = setattr(snd, 'energy', 4500)` is kind of awkward?

Reply:
Yeah I initially had it that way but Diling requested it be an attribute. No reason to not just have both though, one for the users and one we use internally. I'll add that in.